src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -50,23 +39,154 @@ static ColliderCylinderInit D_80B18910 = {
+                    this->unk_208++;
+                    break;
+            }
+            return 0;
+        default:
+            return 1;
+    }
+}
+
+void EnTg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 28.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600AE40, &D_06005040, NULL, NULL, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    this->actor.targetMode = 6;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->unk_208 = globalCtx->state.frames % 2;
+    this->actionFunc = EnTg_SpinIfNotTalking;
+}
+
+void EnTg_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnTg_SpinIfNotTalking(EnTg* this, GlobalContext* globalCtx) {
+    if (this->isTalking == 0) {
+        this->actor.shape.rot.y += 0x800;
+    }
+}
+
+void EnTg_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+    s32 pad;
+    f32 temp;
+    Vec3s sp2C;
+
+    sp2C.x = this->actor.world.pos.x;
+    sp2C.y = this->actor.world.pos.y;
+    sp2C.z = (s16)this->actor.world.pos.z + 3;
+    this->collider.dim.pos = sp2C;
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    SkelAnime_Update(&this->skelAnime);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->actionFunc(this, globalCtx);
+    temp = this->collider.dim.radius + 30.0f;
+    func_800343CC(globalCtx, &this->actor, &this->isTalking, temp, func_80B18360, func_80B183F8);
+}
+
+s32 EnTg_OverrideLimbDrawOpa(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             void* thisx) {
+    return 0;
+}
+
+void EnTg_PostLimbDrawOpa(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    EnTg* this = (EnTg*)thisx;
+    Vec3f D_80B18968 = { 0.0, 800.0, 0.0 };
+    if (limbIndex == 9) {
+        Matrix_MultVec3f(&D_80B18968, &this->actor.focus.pos);
+    }
+}
+
+Gfx* func_80B18778(GraphicsContext* gfxCtx, u8 r, u8 g, u8 b, u8 a) {
+    Gfx* displayList;
+
+    displayList = Graph_Alloc(gfxCtx, 2 * sizeof(Gfx));
+    gDPSetEnvColor(displayList, r, g, b, a);
+    gSPEndDisplayList(displayList + 1);
+    return displayList;
+}
+
+void EnTg_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_tg.c", 462);
+    Matrix_Translate(0.0f, 0.0f, -560.0f, MTXMODE_APPLY);
+    gSPSegment(POLY_OPA_DISP++, 0x08, func_80B18778(globalCtx->state.gfxCtx, 0, 0x32, 0xA0, 0));
+    gSPSegment(POLY_OPA_DISP++, 0x09, func_80B18778(globalCtx->state.gfxCtx, 0xFF, 0xFF, 0xFF, 0));
```

Comment:
Explanation: as you documented it, `func_80B18778` takes r,g,b,a as its last parameters and it is a convention in the codebase to use decimal instead of hexadecimal for colors